### PR TITLE
Call aiXplain discover models

### PR DESCRIPTION
**Pull request by lucas-aixplain**
This PR updates the code to call models passing the url and api key, as we already have with pipelines. It is changing the folder structure by having factories, that create model or pipeline objects.

Also, this PR adds methods to call models or pipelines asynchronously.

### CODE REVIEW

File: aixplain/modules/model.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        while not completed and (end - start) < timeout:
+            try:
+                resp = self.poll(poll_url, name=name)
+                completed = resp['completed']
+
+                end = time.time()
+                time.sleep(wait_time)
+                if wait_time < 60:
+                    wait_time *= 1.1
+            except:
```
Comment (krishnadurai) [inline aixplain/modules/model.py:70]: `except Exception as e:`

A plain `except` catches many other undesirable exceptions like Control + C exit as well.

File: aixplain/modules/model.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                logging.error(f"ERROR: polling for {name} ({self.api_key}): Continue")
+                break
+        
+        if resp['completed'] is True:
+            try:
+                logging.info(f"Final status of polling for {name} ({self.api_key}): SUCCESS - {resp}")
+                return True, resp
+            except:
```
Comment (krishnadurai) [inline aixplain/modules/model.py:78]: Same as above

File: aixplain/modules/model.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        ---
+            Output: parsed output from model
+        """
+        start = time.time()
+        try:           
+            poll_url = self.run_async(data, name=name)
+            end = time.time()
+            success, response = self.__polling(poll_url, name=name)
+            return { 'success': success, 'response': response, 'error': None, 'elapsed_time': end - start }
+        except:
```
Comment (krishnadurai) [inline aixplain/modules/model.py:131]: Same as above

**Review by krishnadurai — CHANGES_REQUESTED** [review]
I left a few comments, please address them

File: README.md [hunk 4]
```diff
@@ -16,29 +16,41 @@ The image below shows a sample aiXplain pipeline built for subtitling video file
 ## Installation
 
 ```
-pip install aixplain-pipelines
+pip install aixplain
```
[the remark this answers is not in the document]
Comment (lucas-aixplain) [reply README.md:19]: I was unsure about this because since we're changing the folder name, I thought we also needed to change the package name.

**Review by krishnadurai — CHANGES_REQUESTED** [review]
As discussed several times, let us not name the module 'aixplain_sdk', that would corrupt the namespace for future packages.

File: tests/model_test.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+from dotenv import load_dotenv
+
+load_dotenv()
+
+import time
+from aixtend.utils.config import MODELS_RUN_URL, MODEL_API_KEY
+from aixtend.factories.model_factory import ModelFactory
+
+def test_mt1():
```
Comment (krishnadurai) [inline tests/model_test.py:27]: Call all of these unit tests as functional tests. Mock the backend calls for unit tests.

File: tests/model_test.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    response = model.run(data)
+    assert response['status'] == 'SUCCESS'
+
+def test_mt2():
+    url = MODELS_RUN_URL
+    api_key = MODEL_API_KEY
+
+    model = ModelFactory.initialize(api_key=api_key, url=url)
+
+    data = 'https://aixplain-platform-assets.s3.amazonaws.com/samples/en/bestofyou.txt'
```
Comment (krishnadurai) [inline tests/model_test.py:43]: Can we have a direct file upload for audio and video formats?

**Review by krishnadurai — APPROVED** [review]

**Pull request merged**